File: Ejercicio_4.py
class Motor: 
    def __init__(self, tipo, caballos):
        self.tipo= tipo
        self.caballos= caballos 

    def describir(self): 
        return f"Motor {self.tipo} de {self.caballos} CV"

class Coche: 
    def __init__(self, marca, modelo, motor):
        self.marca= marca 
        self.modelo= modelo
        self.motor= motor

    def mostrar_info(self): 
        print(f"Coche: {self.marca} {self.modelo}")
        print(self.motor.describir())

motor_v8= Motor("gasolina", 450)

auto= Coche("Nissan", "Sentra", motor_v8)

auto.mostrar_info()

File: RA 1.1/Ejercicio_4.cpp
#include <iostream> 
using namespace std; 

class Motor { 
    protected: 
        string tipo; 
        double caballos; 

    public: 
        Motor(string _tipo, double _caballos): tipo(_tipo), caballos(_caballos) {}
        ~Motor() {}

        string describir() { 
            return "Motor "+this->tipo+" de "+to_string(this->caballos)+" CV";
        }
};

class Coche {
    protected: 
        string marca; 
        string modelo; 
        Motor motor;

    public: 
        Coche(string _marca, string _modelo, Motor _motor): marca(_marca), modelo(_modelo),  motor(_motor) {}
        ~Coche() {}

        void mostrar_info() {
            cout<<"Coche: "<<this->marca<<" "<<this->modelo<<endl;
            cout<<this->motor.describir()<<endl;
        }
};

int main() { 
    Motor motor_v8("gasolina", 450); 

    Coche automovil("Nissan", "Sentra", motor_v8); 

    automovil.mostrar_info();

    return 0;
}